### ISSUE TITLE
The model's performance on extreme few-shot settings

### ISSUE THREAD

**Issue by kfxw** [browatbn2/3FabRec#8]
Hi!
I tried to train the model with only 1 training image. I excuted the training script with the command
```shell
python3 train_aae_landmarks.py --sessionname one_shot_aae -r ./ --train-encoder 0 --dataset w300 --train-count 1 --batchsize 1 --epoch 10000 --eval-freq 1000 --save-freq 1000
```
However, there is still a gap between the trained model and the reported performance in the paper. I got nmes=15.20/18.11/15.93 with the above command. Did I miss any important detail when I trained this model? Or is it just simply because the epoch number is not enough?
Thanks!

**Comment by browatbn2** [comment]
Hi,
the nmes you got there, are these the numbers that are output by the script? In that case the 3 numbers are inner/outer/total lms. In the tables in the papers the columns are the dataset splits: common/challenging/full.

By default the evaluation is run on the full dataset. Did you set it to challenging by any chance? Param `--test-split`.

**Comment by kfxw** [comment]
The nmes are directly output by the script. So I suppose they are aligned with the results in paper.
As for the test split, I used the default setting and there are 600+ test images. So it should be the full test set.

**Comment by browatbn2** [comment]
Did you actually use your own (unsupervised) model or the supplied one?

In any case, I just ran it myself and noticed that the data augmentation got lost in the landmark training. I just pushed a fix. 

You can try now with e.g. `--daug 6` and you should be able to get better results. I would also suggest to set `eval-freq` to maybe 100 and keep track of the validation error. The model should converge for one shot in less than 1000 epochs. With too much training there is a danger of overfitting that one training image.

**Comment by kfxw** [comment]
Yes, I used the pretrained aae and actually the full dataset training seemed ok.
I opened up the augmentation 6, the model's performance is improved to some extend, but still lower than the expected one.
The wired thing is that the validation error l_lms didn't go down after around 400 iterations. The gap between the training loss and val loss is significant.
I'll post the log below.

**Comment by kfxw** [comment]
https://github.com/kfxw/tmp/blob/master/test.log

**Comment by browatbn2** [comment]
Ah, ok, you're not actually starting with the pre-trained model. Seems like you're using the wrong model name/path:

```
[2020-09-04 16:14:53] Resuming session test from snapshot ./...
[2020-09-04 16:14:53] Model ./ trained for 0 iterations (0:00:00).
```

By default the model is loaded from ./data/models/snapshots/<modelname>/saae.mdl when you call the script with `-r <modelname>'. 

It should say something like:
`Model lms_wflw trained for 2588294 iterations (7 days, 8:30:25).`

**Comment by kfxw** [comment]
Well, it is actually a mistake I made some time ago. I deleted the content in the meta.json somehow. So the log information is incorrect.
However, if you look at the l_rec, the reconstruction loss behaves normally as it flunctuated around 10~20 and I think the pretrained model is loaded properlly.

**Comment by browatbn2** [comment]
Does the visualization look reasonable? You probably need to run it `--print-freq 1` to see something.

**Comment by kfxw** [comment]
By visualiting the internal results, I find the real cause of my problem.
I found that the only training image from all epochs looked the same, so I further found that the data augmentation, though I added --daug 6 in the command, was actually not added into the dataloader. Then I noticed your commit yesterday and confirmed this cause.
By the way, I think it would be better to leave the data augmentation None for the validation data loader.
```python
transform = ds_utils.build_transform(deterministic=not train, daug=args.daug) if phase == TRAIN else None
```
Thanks a lot for your patient replies!  : )

**Comment by browatbn2** [comment]
Great! Glad it's working now. Thanks for letting me know about the issue, I would not have noticed the missing data augmentation otherwise. :)

I also just uploaded more pre-trained models. You can get them by running `./download_models.sh` again. 

This way it's easier to start with a model that hasn't been trained on WFLW (or other landmarks) yet. Using the current model (lms_wflw) wouldn't really be one-shot/few-shot learning.
